x-pack/plugins/ingest_pipelines/server/routes/api/update.ts

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import { schema } from '@kbn/config-schema';
+
+import { Pipeline } from '../../../common/types';
+import { API_BASE_PATH } from '../../../common/constants';
+import { RouteDependencies } from '../../types';
+
+const bodySchema = schema.object({
+  description: schema.string(),
+  processors: schema.arrayOf(schema.recordOf(schema.string(), schema.any())),
+  version: schema.maybe(schema.number()),
+  onFailure: schema.maybe(schema.arrayOf(schema.recordOf(schema.string(), schema.any()))),
+});
+
+const paramsSchema = schema.object({
+  name: schema.string(),
+});
+
+export const registerUpdateRoute = ({
+  router,
+  license,
+  lib: { isEsError },
+}: RouteDependencies): void => {
+  router.put(
+    {
+      path: `${API_BASE_PATH}/{name}`,
+      validate: {
+        body: bodySchema,
+        params: paramsSchema,
+      },
+    },
+    license.guardApiRoute(async (ctx, req, res) => {
+      const { callAsCurrentUser } = ctx.core.elasticsearch.dataClient;
+      const { name } = req.params as typeof paramsSchema.type;
```

Comment:
We should consider modifying license service `gaurdApiRoute` to preserve type information:

```
  guardApiRoute<P, Q, B>(handler: RequestHandler<P, Q, B>) {
    const license = this;

    return function licenseCheck(
      ctx: RequestHandlerContext,
      request: KibanaRequest<P, Q, B>,
```

Then we can remove `as` declarations (although I can see we may want to keep as `Pipeline` on the body).